jupyterlab_celltests/lint.py

```diff
@@ -7,71 +7,52 @@
 from .shared import extract_extrametadata
 from .define import LintMessage, LintType
 
+# TODO why do these fns take a parameter that's also present in
+# metadata?  (e.g. lint function take lines_per_cell, but that's
+# already in metadata)
+
+# TODO: there's duplication with tests here. Should both use same
+# underlying code. Also, some of these things are not lint (e.g.
+# test coverage, while some of the tests are really just lint).
+
 
 def lint_lines_per_cell(lines_per_cell, metadata):
     ret = []
-    passed = True
-
-    if lines_per_cell:
-        for i, lines_in_cell in enumerate(metadata.get('cell_lines', [])):
-            if lines_in_cell <= lines_per_cell:
-                ret.append(LintMessage(i+1, 'Checking lines in cell', LintType.LINES_PER_CELL, True))
-            else:
-                ret.append(LintMessage(i+1, 'Checking lines in cell', LintType.LINES_PER_CELL, False))
-                passed = False
-    return ret, passed
+    if lines_per_cell < 0:
+        return [], True
```

Comment:
This would just be a silent failure, right? I would either raise an error, or ensure all cells give a lint. Same below.

Reply:
`lines_per_cell` is the specification of max lines allowed per cell. The existing convention in celltests (unrelated to this PR) is that -1 means "don't check lines per cell".

Before this PR, `lines_per_cell` of -1 would (presumably accidentally) result in a lint failure for every cell, because every cell would have > -1 lines in it.

Now, `lines_per_cell` of -1 passes no matter how many lines per cell ("do not check the lines per cell"). Whether -1 is the right way to achieve that is a different question, but I don't think it's related to this PR (as it was already that way).

I believe a similar thing applies to other cases below. 

So, for instance, previously if `class_definitions` (max class definitions per notebook) was set to 0 - which ought to mean no class definitions allowed - then any number of class definitions was permitted (no lint failure). If `class_definitions` was set to -1, any number of actual class definitions (including 0) would fail.

Reply:
You're saying you'd like to see a 'lines per cell lint passed' for every cell, even if the lines per cell lint is not actually being checked? (Whereas in the PR as it is, I have gone for returning an empty list to indicate nothing was checked. I believe/guess that was the intent of the original code, which was returning an empty list if the `if lines_per_cell` test was falsey.)